views/forms/BarChartFields.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useEffect } from 'react'
+import Delete from '@material-ui/icons/Delete';
+import TextField from '@material-ui/core/TextField'
+import Button from '@material-ui/core/Button'
+import Typography from '@material-ui/core/Typography';
+
+const BarChartFields = ({ 
+  classes,
+  updateFormValues,
+  formValues,
+  setFormValues,
+}) => {
+  const { title } = formValues
+
+  const [chartTitle, setChartTitle] = useState('')
+  
+  const addValueAndLabelField = () => setFormValues(prevState => ({
+    ...prevState,
+    fieldLabelValueMap: [...prevState
+      .fieldLabelValueMap,
+        { 
+          value: '', 
+          label: '' 
+        }
+      ]
+    })
+  )
+  const removeValueAndLabelField = (id) => setFormValues((prevState) => ({ 
+    ...prevState,
+    fieldLabelValueMap: [...prevState
+      .fieldLabelValueMap
+      .filter((_, index) => index !== id)
+      ]
```

Comment:
You don't need to use `...` here since filter will give you a new array.

```suggestion
    fieldLabelValueMap: prevState.fieldLabelValueMap.filter((_, index) => index !== id)
```

Reply:
[ab607bc](https://github.com/AMP-SCZ/dpdash/pull/84/commits/ab607bcdc9895d5c9da30b9ab50689aab6b3b00e)